server.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const weather = require('./weather.json');
+
+const app = express();
+
+app.use(cors());
+
+const PORT = process.env.PORT;
+
+app.get('/hello', (request, response) => { response.send('Hey, there!') });
+app.get('/weather', handleGetWeather);
+
+function handleGetWeather(req, res) {
```

Comment:
Hi there!

I know you had some questions about the lab/specifically about lat, lon, searchquery. 
I like your thoughts on creating a function (`function handleGetWeather`) this is good! 

The `req` object has certain properties on it, one being the request query string (look up express req and res).
The first part of card 1 says, "Create an API endpoint of /weather that processes a GET request that contains lat, lon and searchQuery information"

- Inside your function:
   - use req.query.searchQuery
- Card 2 says that you have to use  .find() method to discover which city the lat, lon and searchQuery information belong to."
   - **IF** the user did not search for one of the three cities that we have information about (Seattle, Paris, or Amman), return an error.**
   - You set the json data to the `weather` variable so use: weather.find(city => city.city_name.toLowerCase() === searchQuery.toLowerCase());`
   - If the user did not search for one of three cities(see above instructions from card) return error etc.

This is kinda how you get this going. Take things one step at a time, task by task and you will good. Grab a TA to get some help on this too 👍

Reply:
thank you, Riva. I will get into it today.